test/linalg/symmetric.jl

```diff
@@ -42,206 +41,229 @@ end
-        end
-
-        eltya == BigFloat && continue # Revisit when implemented in julia
-        d, v = eig(asym)
-        @test asym*v[:,1] ≈ d[1]*v[:,1]
-        @test v*Diagonal(d)*v' ≈ asym
-        @test isequal(eigvals(asym[1]), eigvals(asym[1:1,1:1]))
-        @test abs.(eigfact(Hermitian(asym), 1:2)[:vectors]'v[:,1:2]) ≈ eye(eltya, 2)
-        eig(Hermitian(asym), 1:2) # same result, but checks that method works
-        @test abs.(eigfact(Hermitian(asym), d[1] - 1, (d[2] + d[3])/2)[:vectors]'v[:,1:2]) ≈ eye(eltya, 2)
-        eig(Hermitian(asym), d[1] - 1, (d[2] + d[3])/2) # same result, but checks that method works
-        @test eigvals(Hermitian(asym), 1:2) ≈ d[1:2]
-        @test eigvals(Hermitian(asym), d[1] - 1, (d[2] + d[3])/2) ≈ d[1:2]
-        @test full(eigfact(asym)) ≈ asym
-        @test eigvecs(Hermitian(asym)) ≈ eigvecs(asym)
-
-        # relation to svdvals
-        @test sum(sort(abs.(eigvals(Hermitian(asym))))) == sum(sort(svdvals(Hermitian(asym))))
-
-        # cond
-        @test cond(Hermitian(asym)) ≈ cond(asym)
-
-        # det
-        @test det(asym) ≈ det(Hermitian(asym, :U))
-        @test det(asym) ≈ det(Hermitian(asym, :L))
-        if eltya <: Real
-            @test det(asym) ≈ det(Symmetric(asym, :U))
-            @test det(asym) ≈ det(Symmetric(asym, :L))
-        end
-        @test det(a + a.') ≈ det(Symmetric(a + a.', :U))
-        @test det(a + a.') ≈ det(Symmetric(a + a.', :L))
-
-        # isposdef[!]
-        @test isposdef(Symmetric(asym)) == isposdef(full(Symmetric(asym)))
-        @test isposdef(Hermitian(asym)) == isposdef(full(Hermitian(asym)))
-        if eltya != Int
-            @test isposdef!(Symmetric(copy(asym))) == isposdef(full(Symmetric(asym)))
-            @test isposdef!(Hermitian(copy(asym))) == isposdef(full(Hermitian(asym)))
-        end
+            @testset "conversion" begin
+                @test Symmetric(asym) == convert(Symmetric,Symmetric(asym))
+                if eltya <: Real && eltya != Int
+                    typs = [Float16,Float32,Float64]
+                    for typ in typs
+                        @test Symmetric(convert(Matrix{typ},asym)) == convert(Symmetric{typ,Matrix{typ}},Symmetric(asym))
+                    end
+                end
+                if eltya <: Complex && eltya != Int
+                    typs = [Complex64,Complex128]
+                    for typ in typs
+                        @test Hermitian(convert(Matrix{typ},asym)) == convert(Hermitian{typ,Matrix{typ}},Hermitian(asym))
+                    end
+                end
+            end
 
-        # rank
-        let A = a[:,1:5]*a[:,1:5]'
-            # Make sure A is Hermitian even in the present of rounding error
-            # xianyi/OpenBLAS#729
-            A = (A' + A) / 2
-            @test rank(A) == rank(Hermitian(A))
-        end
+            @testset "issymmetric, ishermitian" begin
+                if eltya <: Real
+                    @test issymmetric(Symmetric(asym))
```

Comment:
This test will work for non-real case too?
and `@test ishermitian(Hermitian(aherm))` should also work in all cases.